src/commands/generate/template.ts

```diff
@@ -0,0 +1,276 @@
+import { Flags, CliUx } from '@oclif/core';
+import Command from '../../base';
+// eslint-disable-next-line
+// @ts-ignore
+import AsyncAPIGenerator from '@asyncapi/generator';
+import path from 'path';
+import { load, Specification } from '../../models/SpecificationFile';
+import { watchFlag } from '../../flags';
+import { isLocalTemplate, Watcher } from '../../utils/generator';
+import { ValidationError } from '../../errors/validation-error';
+
+import type { Example } from '@oclif/core/lib/interfaces';
+
+const red = (text: string) => `\x1b[31m${text}\x1b[0m`;
+const magenta = (text: string) => `\x1b[35m${text}\x1b[0m`;
+const yellow = (text: string) => `\x1b[33m${text}\x1b[0m`;
+const green = (text: string) => `\x1b[32m${text}\x1b[0m`;
+
+interface ParsedFlags {
+  params: Record<string, string>,
+  disableHooks: Record<string, string>,
+  mapBaseUrlToFolder: {
+    url: string,
+    folder: string
+  }
+}
+
+export default class Template extends Command {
+  static description = 'generate whatever you want using templates compatible with AsyncAPI Generator';
+
+  static examples: Example[] = [
+    'asyncapi generate asyncapi.yaml @asyncapi/html-template'
+  ];
+
+  static flags = {
+    help: Flags.help({ char: 'h' }),
+    'disable-hook': Flags.string({
+      char: 'd',
+      description: 'disable a specific hook type or hooks from a given hook type',
+      multiple: true
+    }),
+    install: Flags.boolean({
+      char: 'i',
+      default: false,
+      description: 'installs the template and its dependencies (defaults to false)'
+    }),
+    debug: Flags.boolean({
+      description: 'enable more specific errors in the console'
+    }),
+    'no-overwrite': Flags.string({
+      char: 'n',
+      multiple: true,
+      description: 'glob or path of the file(s) to skip when regenerating'
+    }),
+    output: Flags.string({
+      char: 'o',
+      description: 'directory where to put the generated files (defaults to current directory)',
+      required: true
+    }),
+    'force-write': Flags.boolean({
+      default: false,
+      description: 'force writing of the generated files to given directory even if it is a git repo with unstaged files or not empty dir (defaults to false)'
+    }),
+    'watch-template': watchFlag(
```

Comment:
shouldn't it be watch as it is globally in this CLI?

Reply:
We are using the global flag https://github.com/asyncapi/cli/blob/master/src/flags.ts, do you mean the message should we use `Enable Watch Mode`.

Reply:
I mean the name of the flag, I think it should be `--watch`

Reply:
it is ok to change flag names as we are migrating and basically creating kinda new CLI, so IMHO better to be consistent with AsyncAPI CLI